src/mlagility/api/model_api.py

```diff
@@ -116,10 +123,71 @@ def benchmark_model(
                 )
                 perf = cpu_model.benchmark(backend=backend)
 
+        elif device == "x86" and runtime in ["torch", "torch_compiled"]:
+
+            # Create cache folder with stats file
+            # Although building with an empty sequence is possible, we don't want
+            # the model to be rebuilt when other devices are used, causing the
+            # results inside of the stats file to be lost.
+            build_model(
+                model=model,
+                inputs=inputs,
+                build_name=build_name,
+                cache_dir=cache_dir,
+                rebuild=rebuild,
+                sequence=sequence,
+            )
+
+            torch_version = torch.__version__
+            if runtime == "torch_compiled":
+                clean_torch_version = torch_version.split("+")[0]
+                if version.parse(clean_torch_version) < version.parse("2.0.0"):
+                    BenchmarkException(
+                        (
+                            f"{runtime} can only be used with Pytorch 2.0.0 or above. "
+                            f"However, version {torch_version} was found."
+                        )
+                    )
+
+                model = torch.compile(model)
+
+            if not build_only:
+                repetitions = 100
+                total_time = [0] * repetitions
+                for idx in range(repetitions):
+                    start_time = time.process_time()
+                    model(**inputs)
+                    end_time = time.process_time()
```

Comment:
Is `time.process_time()` the right choice here?

Reply:
`time.process_time()` measures the CPU time used by the process. I believe we want to measure the wall clock time here. So `time.perf_counter()` might be a better option.

Reply:
We should be consistent with our time measurement. The ORT and TRT measurement code uses a different library (`timeit`): https://github.com/groq/mlagility/blob/1bef1c18720c758f32ae68a897a5f52687f6fe16/src/mlagility/api/run_ort_model.py#L25

Suggest snapping this function to use `timeit` as well, but you could also change the ORT and TRT code instead.

Reply:
Using `timeit` for consistency